examples/mssn.py

```diff
@@ -63,46 +52,52 @@
 # generated input signal x (x_train and x_test).
 
 # define sets of internal, external and ground nodes
-ctx  = np.load(os.path.join(DATA_DIR, 'cortical.npy'))
+ctx = iodata.load_file('cortical.npy')
 
+# number of features in task data
 n_features = x_train.shape[1]
-nodes = np.arange(n_reservoir_nodes)
-gr_nodes  = np.random.choice(np.where(ctx == 1)[0], 1) # we select a single random ground node from cortical regions
-ext_nodes = np.random.choice(np.where(ctx == 0)[0], n_features) # we select a random set of input nodes from subcortical regions
-int_nodes = np.setdiff1d(nodes, np.union1d(gr_nodes,ext_nodes)) # we use the reamining cortical regions as output nodes
 
-# However, because not all subcortical nodes are used as input nodes in this case, we create 
-# a new set of output nodes that include (only) all cortical regions but those corresponding 
-# to grounded nodes
-output_nodes = np.setdiff1d(np.where(ctx == 1)[0], gr_nodes) # nodes actually used to perform the task
+# we select a single random ground node from cortical regions
+gr_nodes = conn.get_nodes(
+    'random', nodes_from=conn.get_nodes('ctx'), n_nodes=1)
+
+# we use the remaining cortical regions as output nodes
+output_nodes = conn.get_nodes('ctx', nodes_without=gr_nodes)
+
+# we select external nodes as random set of input nodes from subcortical regions
+ext_nodes = conn.get_nodes(
+    'random', nodes_from=conn.get_nodes('subctx'), n_nodes=n_features)
+
+# we select internal nodes as all nodes except of ground nodes external nodes
+int_nodes = conn.get_nodes(
+    'all', nodes_without=np.union1d(gr_nodes, ext_nodes))
 
 # We will use resting-state networks as readout modules. These intrinsic networks
 # define different sets of output nodes
-rsn_mapping = np.load(os.path.join(DATA_DIR, 'rsn_mapping.npy'))
-rsn_mapping = rsn_mapping[output_nodes] # we select the mapping only for output nodes 
+rsn_mapping = iodata.load_file('rsn_mapping.npy')
+# we select the mapping only for output nodes
+rsn_mapping = rsn_mapping[conn.idx_node][output_nodes]
 
 # evaluate network performance across various dynamical regimes
-# we do so by varying the value of alpha 
-import pandas as pd
-from conn2res import reservoir, coding
+# we do so by varying the value of alpha
 
-alphas = np.linspace(0,2,11)[1:]
+alphas = np.linspace(0, 2, 11)[1:]
 df_subj = []
 for alpha in alphas:
-    
+
     print(f'\n----------------------- alpha = {alpha} -----------------------')
 
     # instantiate an Memristive Network object
-    MMN = reservoir.MSSNetwork(w=alpha * conn.copy(),
+    MMN = reservoir.MSSNetwork(w=alpha * conn.w,

```

Comment:
As I mentioned in my generic intro of this feature branch, here I pass conn.w to the reservoir but if you want to perform operations on the connectivity matrix within the reservoir as it is the case for the functions check_symmetric, make_symmetric etc, you will need to move those functions inside the Conn class and pass conn here.